### DISCUSSION TITLE
How do we get SupabaseAuth User object via JWT Token?

### DISCUSSION THREAD

**Discussion by rphly** [supabase/supabase#1582] [Questions] [answered]
I'm using Supabase on server side and was wondering how I might go about retrieving the Supabase Auth user object using the token provided during login / verification?

**Comment by awalias** [comment, marked as answer]
we [added a method](https://github.com/supabase/gotrue-js/blob/4312f53411dc6fdf963f8adb48200cddf9905bd9/src/GoTrueClient.ts#L288) in gotrue-js for this and we should probably do the same in gotrue-py, issue raised here: https://github.com/supabase/gotrue-py/issues/2

in the meantime:

you can get the user details using https://github.com/supabase/gotrue-py/blob/38eddcb51e8a59e068a358971c09425138dd852e/gotrue/api.py#L160 by passing the access_token (jwt)

you can also parse the JWT itself using the JWT_secret from the supabase dashboard (Settings > API) to get some basic user info

![image](https://user-images.githubusercontent.com/458736/119302597-f49d5500-bc96-11eb-9841-9aa2400eb2a5.png)


to start requesting data specific to that user however this access_token needs to be set in the `Authorization: Bearer <access_token>` header when requesting data from supabase. It may be possible to manually set it inside Supabase-py, it's certainly possible to set the header inside Gotrue-py (which is the Auth library wrapped by Supabase-py)

alternatively you may need to ask the user to SignIn() after they've confirmed their email

hope this helps!
